spring-cloud-azure-feature-management/src/main/java/com/microsoft/azure/spring/cloud/feature/manager/feature/filters/PercentageFilter.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.feature.manager.feature.filters;
+
+import static com.microsoft.azure.spring.cloud.feature.manager.FilterParameters.PERCENTAGE_FILTER_SETTING;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.microsoft.azure.spring.cloud.feature.manager.FeatureFilter;
+import com.microsoft.azure.spring.cloud.feature.manager.entities.FeatureFilterEvaluationContext;
+
+/**
+ * A feature filter that can be used to activate a feature based on a random percentage.
+ */
+@Component("PercentageFilter")
+public class PercentageFilter implements FeatureFilter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PercentageFilter.class);
+
+    /**
+     * Performs a percentage based evaluation to determine whether a feature is enabled.
+     * @param context The feature evaluation context.
+     * @return True if the feature is enabled, false otherwise.
+     */
+    @Override
+    public boolean evaluate(FeatureFilterEvaluationContext context) {
+        try {
+            Thread.sleep(5000);
```

Comment:
Wondering why it gonna sleep for 5 sec here?

Reply:
It must have been from a test and missed. Will remove.